internal/configs/virtualserver_test.go

```diff
@@ -895,6 +925,33 @@ func TestGenerateProxyPassProtocol(t *testing.T) {
 	}
 }
 
+func TestGenerateString(t *testing.T) {
+	tests := []struct {
+		nextUpstream string
```

Comment:
the signature of the function under test is `func generateString(s string, defaultS string) string {`
however, in the test you use `nextUpstream` variable. Also, because the values are related to a specific NGINX directive, this relationship can confuse the reader.

Consider:
- renaming `nextUpstream` to something like `inputString` or other thing related to the function signature 
- using non NGINX directives values like `foo` and `abc`. 

Also, the third test case look very similar to the first one. I think it is redundant. Although if removing one of them, in the remaining please make sure that inputS and default has different values.